### PR TITLE
Duck DNS alias support

### PR DESCRIPTION
**Pull request by jmorahan**
For when you have another domain CNAME'd to the Duck DNS one.

Context: https://community.letsencrypt.org/t/dont-know-what-content-my-text-record-should-have/85439/2?u=jmorahan

The config would look like:

```
{
  "lets_encrypt": {
    "accept_terms": true,
    "certfile": "fullchain.pem",
    "keyfile": "privkey.pem"
  },
  "token": "sdfj-2131023-dslfjsd-12321",
  "domains": ["subdomain.example.com"],
  "aliases": {
    "subdomain.example.com": "example.duckdns.org"
  },
  "seconds": 300
}
```

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @jmorahan, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;hassio-addons#521).

Once you do that we will be able to review and accept this pull request.

Thanks!

**Comment by pvizeli** [comment]
Is that tested? because a config struct like this will be not supported. Maybe:
```json
{
  "aliases": [
     {
        "domain": "my.duckdns.org",
        "alias": "subdomain.xy.com"
     }
  ]
}
```

**Comment by jmorahan** [comment]
Hi, I don't have a hass.io dev environment set up so I couldn't test it end to end, sorry. Didn't realise there was a config schema. Lemme see if I can fix it.

**Comment by jmorahan** [comment]
Hope that's better, though again I haven't been able to fully test it myself. Working on that...

**Comment by pvizeli** [comment]
I try it out tonight

**Comment by thib5** [comment]
Hi ! they update the add-on to make it possible with alias … but I’m not able to make it work… this is my config

{
“lets_encrypt”: {
“accept_terms”: true,
“certfile”: “fullchain.pem”,
“keyfile”: “privkey.pem”
},
“token”: “BLAHBLAHBLAH”,
“domains”: [
“my.duckdns.org”
],
“aliases”: [
“myowndomain.com”
],
“seconds”: 300
}

**Comment by jamesking420** [comment]
I don't think the change is live yet.

**Comment by Fma965** [comment]
Any update on this? from what i can see all changes have been approved and passed checks, just conflicts with changelog which is obviously not an issue.

**Comment by thib5** [comment]
also what is the exact way to add it once the change will be live ?

**Comment by jamesking420** [comment]
Does anyone know if this will be integrated?

**Comment by jamesking420** [comment]
Looks like "won't fix"? But an the info for fixing it is there?

**Comment by jamesking420** [comment]
Is this going to be merged?

File: duckdns/config.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
     },
     "token": null,
     "domains": [null],
+    "aliases": [null],
```
Comment (pvizeli) [inline duckdns/config.json]: should be not null

**Comment by dpDesignz** [comment]
Has this been implemented? I only want to use DuckDNS as a backup to my own domains, but I need to have the SSL installed for them as well.

**Comment by pvizeli** [comment]
Maybe someone will pick that up and rebase it

**Comment by jamesking420** [comment]
I don't think this is implemented yet? Is it going to be?

**Comment by teixeluis** [comment]
This issue should be picked up as it is quite a relevant use case.

**Comment by noamokman** [comment]
For anyone interested, I have rebased this PR in #1253 😄

**Pull request closed without merging**